### PR TITLE
ROX-13020: Propagate pagination information in SAC search helper

### PR DESCRIPTION
**Pull request by rhybrillou**
## Description

Performance test for scoped access control highlight that with SAC some paginated queries actually scan the whole dataset in postgres mode.
Looking closer at the search helper in postgres mode, it seems pagination data is not propagated properly.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
~~- [ ] Evaluated and added CHANGELOG entry if required~~
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI should be sufficient.

### CODE REVIEW

File: pkg/sac/search_helper.go [hunk 1]
```diff
@@ -211,46 +212,45 @@ func (h *pgSearchHelper) FilteredSearcher(searcher blevesearch.UnsafeSearcher) s
...
-		sacQueryFilter, err = BuildClusterLevelSACQueryFilter(effectiveaccessscope)
+		sacQueryFilter, err = BuildClusterLevelSACQueryFilter(effectiiveAccessScope)
 		if err != nil {
 			return nil, err
 		}
 	default:
-		sacQueryFilter = nil
+		return nil, errors.Errorf("Invalid scope %v for resource %v", h.resourceMD.GetScope(), h.resourceMD)
 	}
 	scopedQuery := search.ConjunctionQuery(q, sacQueryFilter)
```
[the remark this answers is not in the document]
Comment (rhybrillou) [reply pkg/sac/search_helper.go:234]: I opened https://issues.redhat.com/browse/ROX-13374 for an attempt at a generic problem resolution.
The suggested function is indeed a good idea.

**Review by mtodor — APPROVED** [review]
It looks good. Nice work!
I'm interested in perf improvements.

**Pull request merged**